tezos/new_context/src/tezedge_context.rs

```diff
@@ -625,26 +697,42 @@ impl ShellContextApi for TezedgeContext {
         message: String,
         date: i64,
     ) -> Result<ContextHash, ContextError> {
-        // FIXME: does more work than needed, just calculate the hash, no batch
         let date: u64 = date.try_into()?;
-        let (commit_hash, _batch, _referenced_older_entries) =
-            self.tree
-                .prepare_commit(date, author, message, self.parent_commit_hash)?;
-        let commit_hash = ContextHash::try_from(&commit_hash[..])?;
+        let mut repository = self.index.repository.write()?;
+
+        let (commit_hash_id, batch, _referenced_older_entries) = self.tree.prepare_commit(
+            date,
+            author,
+            message,
+            self.parent_commit_hash,
+            &mut *repository,
+        )?;
+
+        repository.write_batch(batch)?;
+        repository.put_context_hash(commit_hash_id)?;
+        let commit_hash = self.get_commit_hash(commit_hash_id, &*repository)?;
 
         Ok(commit_hash)
     }
 
     fn get_last_commit_hash(&self) -> Result<Option<Vec<u8>>, ContextError> {
-        Ok(self.parent_commit_hash.map(|x| x.to_vec()))
+        let repository = self.index.repository.read()?;
+
+        let value = match self.parent_commit_hash {
+            Some(hash_id) => repository.get_value(hash_id)?,
+            None => return Ok(None),
+        };
+
+        Ok(value.map(|v| v.to_vec()))
     }
 
     fn get_merkle_stats(&self) -> Result<MerkleStoragePerfReport, ContextError> {
         Ok(MerkleStoragePerfReport::default())
     }
 
     fn get_memory_usage(&self) -> Result<usize, ContextError> {
-        Ok(self.index.repository.read()?.total_get_mem_usage()?)
+        todo!()
```

Comment:
Make this return `Ok(0)` instead and leave a `TODO`, we should avoid anything that could kill the process.